### PR TITLE
PARQUET-1851: fix parquet metadata converter NPE

### PR DESCRIPTION
**Pull request by chenjunjiedada**
This fixes the NPE as below:

```
java.lang.NullPointerExceptionjava.lang.NullPointerException at
org.apache.parquet.format.converter.ParquetMetadataConverter.addRowGroup(ParquetMetadataConverter.java:476) at
org.apache.parquet.format.converter.ParquetMetadataConverter.toParquetMetadata(ParquetMetadataConverter.java:177) at
org.apache.parquet.hadoop.ParquetFileWriter.serializeFooter(ParquetFileWriter.java:914) at
org.apache.parquet.hadoop.ParquetFileWriter.end(ParquetFileWriter.java:864) at
org.apache.iceberg.parquet.ParquetWriter.close(ParquetWriter.java:206) at
org.apache.iceberg.data.TestLocalScan.writeFile(TestLocalScan.java:429)
```

### CODE REVIEW

**Review by shangxinli — APPROVED** [review]
Good catch!

**Comment by chenjunjiedada** [comment]
@gszadovszky Thanks for the comment and it makes sense to me. How about throwing an exception when detecting the row group's count is zero? Does that sound reasonable to you?

PS. the space changes in the PR is automatically fixed by idea IDE. If you don't like it, I can handle it.

File: parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java [hunk 1]
```diff
@@ -860,6 +860,10 @@ public void endColumn() throws IOException {
    * @throws IOException if there is an error while writing
    */
   public void endBlock() throws IOException {
+    if (currentRecordCount == 0) {
+      throw new IOException("End block with zero record");
```
Comment (gszadovszky) [inline parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java:864]: I would suggest using a `ParquetEncodingException` instead.

**Review by gszadovszky — CHANGES_REQUESTED** [review]
Thanks a lot for the changes. Please find my only comment in the code. Otherwise it looks good to me.

File: parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java (hunk 1 above)
Comment (chenjunjiedada) [reply parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java]: SGTM.

**Review by gszadovszky — APPROVED** [review]

File: parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java (hunk 1 above)
Comment (vdiravka) [reply parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java]: The empty parquet files was used in Drill earlier and now this change breaks the possibility to store empty tables (with header schema only) in parquet files. Any suggestions to bypass this and integrate Parquet-1.12.0 to Drill?

Comment (gszadovszky) [reply parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java]: @vdiravka, you might want to create a separate jira about this topic so we can discuss it in a more open way. Please, also describe what the empty parquet files are used for.

Comment (vdiravka) [reply parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java]: @gszadovszky Done
https://issues.apache.org/jira/browse/PARQUET-2026

**Pull request merged**